appium/webdriver/webdriver.py

```diff
@@ -200,32 +213,34 @@ class WebDriver(
     Sms,
     SystemBars,
 ):
-    def __init__(
+    def __init__(  # noqa: PLR0913
```

Comment:
is it possible to extract the custom pre-init code into a class or static helper method, so we don't need to ignore the linter warning?

Reply:
Do you mean something like:

```
def __init__(self, **kwargs):
    args = getSomething(**kwargs)
    command_executor = args.command_executor
    ...
```
?

I was thinking of defining a custom client config after this PR to drop existing ones such as keep_alive, direct_connect and strict_ssl like `AppiumClientConfig` and:
```
def __init__(
  self,
  command_executor = 'http://127.0.0.1:4444/wd/hub',
  extensions = None,
  client_config = AppiumClientConfig
)
```

with major version update (as a breaking change.) So this lint ignore is a temporary one.

Reply:
the original linter error is about super init method is not the first call in the class constructor. In order to address that we would need to move the preceeding code into separate method call(s). for example:

```
def __init__(self, x, y):
    z = ... # lint warning here
    super().__init__(z=z)
```

could be changed to


```
def __init__(self, x, y):
    super().__init__(z=self.calculate_z(x, y)) # no warning

@statsicmethod
def calculate_z(x, y):
    return z
```

Reply:
oh, i see

Reply:
hm, the error I suppressed was:
```
appium/webdriver/webdriver.py:216:9: PLR0913 Too many arguments in function definition (7 > 6)
    |
214 |     SystemBars,
215 | ):
216 |     def __init__(
    |         ^^^^^^^^ PLR0913
217 |         self,
218 |         command_executor: Union[str, AppiumConnection] = 'http://127.0.0.1:4444/wd/hub',
    |

Found 1 error.
```
so the number of args for the init itself...?


I did very roughly as below but this didn't help:
```
def get_config(command_executor, keep_alive, strict_ssl, client_config):
    if isinstance(command_executor, str):
        client_config = client_config or ClientConfig(
            remote_server_addr=command_executor, keep_alive=keep_alive, ignore_certificates=not strict_ssl
        )
        client_config.remote_server_addr = command_executor
        return client_config
    return None

def get_command_executor(command_executor, keep_alive, strict_ssl, client_config):
    if isinstance(command_executor, str):
        client_config = client_config or ClientConfig(
            remote_server_addr=command_executor, keep_alive=keep_alive, ignore_certificates=not strict_ssl
        )
        client_config.remote_server_addr = command_executor
        return AppiumConnection(remote_server_addr=command_executor, client_config=client_config)
    return command_executor

class WebDriver(
    ...
):
    def __init__(  # noqa: PLR0913
        self,
        command_executor: Union[str, AppiumConnection] = 'http://127.0.0.1:4444/wd/hub',
        keep_alive: bool = True,
        direct_connection: bool = True,
        extensions: Optional[List['WebDriver']] = None,
        strict_ssl: bool = True,
        options: Union[AppiumOptions, List[AppiumOptions], None] = None,
        client_config: Optional[ClientConfig] = None,
    ):
        super().__init__(
            command_executor=get_command_executor(command_executor, keep_alive, strict_ssl, client_config),
            options=options,
            locator_converter=AppiumLocatorConverter(),
            web_element_cls=MobileWebElement,
            client_config=get_config(command_executor, keep_alive, strict_ssl, client_config),
        )

Reply:
I thought this warning is related to a different cause. Please keep the `noqa` then. Nevertheless I like the above change. Please do not forget to add type declarations to new functions